### DISCUSSION TITLE
Multiple links for external_link (config-metadata.csv)

### DISCUSSION THREAD

**Discussion by gpark47** [CollectionBuilder/collectionbuilder.github.io#78] [Q&A] [answered]
Hi there!

I currently have a field titled "Permalink(s)" for each item, that contains 1-3 external links from my data. Based on how it is right now, there is no way to separate different links and only one is possible; it joins everything in that field together as one clickable, broken link. Is there a way or a type of notation that needs to be used in the data, in order to separate multiple external links?

Thank you.

**Comment by evanwill** [comment, marked as answer]
Hi @gpark47 ! 
Sorry, there isn't currently a built in way to do that--the `external_link` option assumes the metadata value is *only* a single url, basically for the type of value like URI's such as rightsstatement.org. 
However, it would be pretty straightforward to add the functionality following the convention you mention--If you point me to your project I could tell you where to customize it!
Best,

**Reply by gpark47** [reply]
Hello! Here is the link to the project: https://github.com/gpark47/filmhistory-gh

**Reply by evanwill** [reply]
@gpark47 since you are using CB-GH, you would have to modify the file ["_includes/js/item-js.html"](https://github.com/gpark47/filmhistory-gh/blob/main/_includes/js/item-js.html) which generates the item pages for CB-GH collections. It's not the easiest file to look at, because it is a big mix of Liquid and js that creates html for the item page. 

I opened a [pull request](https://github.com/gpark47/filmhistory-gh/pull/1) on your project to add the customization--but I just wanted to document what I did here, in case anyone else finds this and would like to do the same tweak. Let me know how it goes!

To tweak how the "external_link" option is handled, we will have to modify the lines 89 - 91 "_includes/js/item-js.html", which look like:

```
{%- else -%}
if (record[{{ f.field | jsonify }}]) { fields += '<dt class="field">{{ f.display_name }}:</dt> <dd class="field-value">{% if f.external_link == "true" %}<a href="' + record[{{ f.field | jsonify }}] + '" target="_blank" rel="noopener">' + record[{{ f.field | jsonify }}] + '</a>{% else %}' + record[{{ f.field | jsonify }}] + '{% endif %}</dd>'; }
{% endif %}{% endfor %}
```

Replace those lines with:

```
/* customized multiple external links */
{%- elsif f.external_link == "true" -%}
        if (record[{{ f.field | jsonify }}]) { 
            var links = record[{{ f.field | jsonify }}].split(';');
            var externalLinks = "";
            for (var i = 0, len = links.length; i < len; i++) {
                if (links[i] != "") {
                    externalLinks += '<a href="' + links[i].trim() + '" target="_blank" rel="noopener">' + links[i].trim() + '</a><br>';
                }
            }
            fields += '<dt class="field">{{ f.display_name }}:</dt> <dd class="field-value">' + externalLinks + '</dd>';
        }
{%- else -%}
if (record[{{ f.field | jsonify }}]) { fields += '<dt class="field">{{ f.display_name }}:</dt> <dd class="field-value">' + record[{{ f.field | jsonify }}] + '</dd>'; }
{% endif %}{% endfor %}
```

**Reply by gpark47** [reply]
Thank you so much for the solution and the explanation. It's working great!
